shell.nix

```diff
@@ -0,0 +1,46 @@
+let
+  here = toString ./.;
+  moz_overlay = import (builtins.fetchTarball
+    "https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz");
+  pkgs = import <nixpkgs> { overlays = [ moz_overlay ]; };
+  rust = (pkgs.rustChannelOf {
+    channel = "nightly";
+  }).rust.override {
+    extensions = [ "rust-src" "rust-analysis" ];
+    targets = [ "thumbv7em-none-eabihf" ];
+  };
+  rustPlatform = pkgs.makeRustPlatform {
+    rustc = rust;
+    cargo = rust;
+  };
+
+
+
+  my-python-packages = p: with p; [
+    pyserial
+  ];
+
+  systemDeps = with pkgs; [
+    udev
+  ];
+  systemLibStr = pkgs.lib.makeLibraryPath systemDeps;
+
+in pkgs.mkShell {
+  packages = [
+    (pkgs.python3.withPackages my-python-packages)
+  ];
+
+  buildInputs = [
+    rust
+    pkgs.gcc-arm-embedded
```

Comment:
These crates don't have any direct dependency on `gcc` (though we do need to use the public Flipper Zero toolchain for the headers provided in the toolchain's sysroot). As such, do we need to pull ARM GCC into the environment?